src/Features/Core/Portable/NavigateTo/INavigateToSearchService.cs

```diff
@@ -1,16 +1,33 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System.Collections.Generic;
+using System;
 using System.Collections.Immutable;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Host;
 
 namespace Microsoft.CodeAnalysis.NavigateTo
 {
+    [Obsolete("Use " + nameof(INavigateToSearchService_RemoveInterfaceAboveAndRenameThisAfterInternalsVisibleToUsersUpdate) + " instead.")]
```

Comment:
So I'm not sure that's really the right message here: we sure don't want people moving to this new interface *either*...at least not with that name!

Reply:
This order allows the new interface to be implemented independently (the hard/slow part), then we can coordinate with all the repositories on an interface rename (the easy part).

Reply:
I guess I assumed that our path forward would be to directly work forward having people consume the existing name with new surface area, rather than moving to the new name and then moving back.

In any case "above" is a pretty meaningless name then for an external contributor to consume.